src/MUIDataTable.js

```diff
@@ -413,7 +416,13 @@ class MUIDataTable extends React.Component {
       }
 
       if (column.filterOptions) {
-        filterData[colIndex] = cloneDeep(column.filterOptions);
+        if (Array.isArray(column.filterOptions)) {
+          filterData[colIndex] = cloneDeep(column.filterOptions);
+          console.error(
+            'Deprecated: filterOptions must now be an object. see https://github.com/gregnb/mui-datatables/tree/master/examples/customize-filter example',
+          );
+        }
+        else filterData[colIndex] = cloneDeep(column.filterOptions.names);
```

Comment:
using `else if else if (column.filterOptions.names)` here will allow us to keep functionality even when names is not provided.

Reply:
ok